src/plugins/data/common/search/expressions/eql_raw_response.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ExpressionTypeDefinition } from '../../../../expressions/common';
+import { EqlSearchStrategyResponse } from '..';
+
+const name = 'eql_raw_response';
+
+export interface EqlRawResponse {
+  type: typeof name;
+  body: EqlSearchStrategyResponse['rawResponse']['body'];
+}
+
+export type EqlRawResponseExpressionTypeDefinition = ExpressionTypeDefinition<
+  typeof name,
+  EqlRawResponse,
+  EqlRawResponse
+>;
+
+interface EqlReponseEvent {
+  _source?: Record<string, unknown>;
+  fields?: Record<string, unknown>;
+}
+
+interface EqlResponseSequence {
+  events: EqlReponseEvent[];
+}
+
+interface EqlResponseBody {
+  hits: {
+    sequences?: EqlResponseSequence[];
+    events: EqlReponseEvent[];
+  };
+}
+
+const flatten = (obj: Record<string, unknown>) => {
```

Comment:
We already have the [`flat`](https://www.npmjs.com/package/flat) package in our `yarn.lock`. I think we can safely list that in the `package.json` and delegate that job to a third-party library.

Reply:
its not a regular flatten, it has some special requirements (like not flattening out arrays)